pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp

```diff
@@ -445,6 +447,58 @@ template <class Precision> class StateVectorKokkos {
         }
     };
 
+    /**
+     * @brief Utility method for samples.
+     *
+     * @param num_samples Number of Samples
+     *
+     * @return Kokkos::View<size_t *> to the samples.
+     * Each sample has a length equal to the number of qubits. Each sample can
+     * be accessed using the stride sample_id*num_qubits, where sample_id is a
+     * number between 0 and num_samples-1.
+     */
+
+    auto generate_samples(size_t num_samples) -> std::vector<size_t> {
+
+        const size_t num_qubits = getNumQubits();
+
+        const size_t N = getLength();
+
+        Kokkos::View<Kokkos::complex<Precision> *> arr_data = getData();
+
+        Kokkos::View<Precision *> probabilities("probabilities", N);
+
+        Kokkos::View<size_t *> samples("num_samples", num_samples * num_qubits);
```

Comment:
```suggestion
        const size_t num_qubits = getNumQubits();
        const size_t N = getLength();

        Kokkos::View<Kokkos::complex<Precision> *> arr_data = getData();
        Kokkos::View<Precision *> probabilities("probabilities", N);
        Kokkos::View<size_t *> samples("num_samples", num_samples * num_qubits);
```

While whitespace is free, we don't need newlines after every variable :)